app/javascript/article-form/components/ArticleCoverImage.jsx

```diff
@@ -49,6 +51,45 @@ export class ArticleCoverImage extends Component {
     });
   };
 
+  useNativeUpload = () => {
+    return Runtime.isNativeIOS('imageUpload');
+  };
+
+  initNativeImagePicker = (e) => {
+    e.preventDefault();
+    window.webkit.messageHandlers.imageUpload.postMessage({
+      id: 'native-cover-image-upload-message',
+    });
+  };
+
+  handleNativeMessage = (e) => {
```

Comment:
Missing tests (blocking): It should be possible to add tests for this code by firing change events on the input it's bound to. Happy to help if you're not sure how to approach this.